src/main/java/appeng/fluids/parts/PartFluidExportBus.java

```diff
@@ -0,0 +1,185 @@
+	@PartModels
+	public static final IPartModel MODELS_HAS_CHANNEL = new PartModel( MODEL_BASE, new ResourceLocation( AppEng.MOD_ID, "part/fluid_export_bus_has_channel" ) );
+
+	private final IActionSource source;
+
+	public PartFluidExportBus( ItemStack is )
+	{
+		super( is );
+		this.getConfigManager().registerSetting( Settings.REDSTONE_CONTROLLED, RedstoneMode.IGNORE );
+		this.getConfigManager().registerSetting( Settings.FUZZY_MODE, FuzzyMode.IGNORE_ALL );
+		this.getConfigManager().registerSetting( Settings.CRAFT_ONLY, YesNo.NO );
+		this.getConfigManager().registerSetting( Settings.SCHEDULING_MODE, SchedulingMode.DEFAULT );
+		this.source = new MachineSource( this );
+	}
+
+	@Override
+	public TickingRequest getTickingRequest( IGridNode node )
+	{
+		return new TickingRequest( TickRates.FluidExportBus.getMin(), TickRates.FluidExportBus.getMax(), this.isSleeping(), false );
+	}
+
+	@Override
+	public TickRateModulation tickingRequest( IGridNode node, int TicksSinceLastCall )
+	{
+		return this.canDoBusWork() ? this.doBusWork() : TickRateModulation.IDLE;
+	}
+
+	@Override
+	protected boolean canDoBusWork()
+	{
+		return this.getProxy().isActive();
+	}
+
+	@Override
+	protected TickRateModulation doBusWork()
+	{
+		if( !this.canDoBusWork() )
+		{
+			return TickRateModulation.IDLE;
+		}
+
+		final TileEntity te = this.getConnectedTE();
+		if( te != null && te.hasCapability( CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, this.getSide().getFacing() ) )
+		{
+			try
+			{
+				final IFluidHandler fh = te.getCapability( CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, this.getSide().getFacing() );
+				final IMEMonitor<IAEFluidStack> inv = this.getProxy().getStorage().getInventory( this.getChannel() );
+				if( fh != null )
+				{
+					for( int i = 0; i < this.getConfig().getSlots(); i++ )
+					{
+						IAEItemStack stack = this.getConfig().getAEStackInSlot( i );
+						if( stack != null && stack.getDefinition() != null )
+						{
+							IFluidHandlerItem ifh = stack.getDefinition().getCapability( CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY, null );
+							if( ifh == null )
+							{
+								throw new NullPointerException( "IFluidHandlerItem is null" );
+							}
+							AEFluidStack toExtract = AEFluidStack.fromFluidStack( ifh.drain( Integer.MAX_VALUE, false ) );
+							toExtract.setStackSize( this.calculateAmountToSend() );
+							IAEFluidStack out = inv.extractItems( toExtract, Actionable.SIMULATE, this.source );
+							if( out != null )
+							{
+								int wasInserted = fh.fill( out.getFluidStack(), true );
+								if( wasInserted > 0 )
+								{
+									inv.extractItems( toExtract, Actionable.MODULATE, this.source );
+									return TickRateModulation.FASTER;
+								}
+							}
+						}
+					}
+					return TickRateModulation.SLOWER;
+				}
+			}
+			catch( GridAccessException e )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)